src/main/java/de/tum/cit/aet/artemis/programming/repository/BuildJobRepository.java

```diff
@@ -39,8 +41,8 @@ public interface BuildJobRepository extends ArtemisJpaRepository<BuildJob, Long>
                 LEFT JOIN Course c ON b.courseId = c.id
             WHERE (:buildStatus IS NULL OR b.buildStatus = :buildStatus)
                 AND (:buildAgentAddress IS NULL OR b.buildAgentAddress = :buildAgentAddress)
-                AND (CAST(:startDate AS string) IS NULL OR b.buildStartDate >= :startDate)
-                AND (CAST(:endDate AS string) IS NULL OR b.buildStartDate <= :endDate)
+                AND (CAST(:startDate AS string) IS NULL OR b.buildSubmissionDate >= :startDate)
```

Comment:
Does it make sense to use the `buildSubmissionDate` to compare with the `startDate` here? Intuitively, I'd think that a running job is included when it started after the `startDate`

Reply:
We need to use `buildSubmissionDate` by submission date to properly show them in the UI because some jobs might not have a build start date. (when they were cancelled before building or when they went missing for some reason).